src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tunnel.py

```diff
@@ -0,0 +1,192 @@
+            logger.info('Websocket, connected status: %s', self.ws.connected)
+            index = index + 1
+            logger.info('Got debugger connection... index: %s', index)
+            debugger_thread = Thread(target=self._listen_to_client, args=(self.client, self.ws, index))
+            web_socket_thread = Thread(target=self._listen_to_web_socket, args=(self.client, self.ws, index))
+            debugger_thread.start()
+            web_socket_thread.start()
+            logger.info('Both debugger and websocket threads started...')
+            logger.warning('Successfully connected to local server..')
+            debugger_thread.join()
+            web_socket_thread.join()
+            logger.info('Both debugger and websocket threads stopped...')
+            logger.warning('Stopped local server..')
+
+    def _listen_to_web_socket(self, client, ws_socket, index):
+        while True:
+            try:
+                logger.info('Waiting for websocket data, connection status: %s, index: %s', ws_socket.connected, index)
+                data = ws_socket.recv()
+                logger.info('Received websocket data: %s, index: %s', data, index)
+                if data:
+                    # Set the response to echo back the recieved data
+                    response = data
+                    logger.info('Sending to debugger, response: %s, index: %s', response, index)
+                    client.sendall(response)
+                    logger.info('Done sending to debugger, index: %s', index)
+                else:
+                    logger.info('Client disconnected!, index: %s', index)
+                    client.close()
+                    ws_socket.close()
+                    break
+            except:
+                traceback.print_exc(file=sys.stdout)
+                client.close()
+                ws_socket.close()
+                return False
+
+    def _listen_to_client(self, client, ws_socket, index):
+        while True:
+            try:
+                logger.info('Waiting for debugger data, index: %s', index)
+                buf = bytearray(4096)
+                nbytes = client.recv_into(buf, 4096)
+                logger.info('Received debugger data, nbytes: %s, index: %s', nbytes, index)
+                if nbytes > 0:
+                    responseData = buf[0:nbytes]
+                    logger.info('Sending to websocket, response data: %s, index: %s', responseData, index)
+                    ws_socket.send_binary(responseData)
+                    logger.info('Done sending to websocket, index: %s', index)
+                else:
+                    logger.warning('Client disconnected %s', index)
+                    client.close()
+                    ws_socket.close()
+                    break
+            except:
+                traceback.print_exc(file=sys.stdout)
+                client.close()
+                ws_socket.close()
+                return False
```

Comment:
Is this boolean value used? Currently, this function only ever returns False or None